lib/jsonwp-proxy/proxy.js

```diff
@@ -155,19 +158,94 @@ class JWProxy {
     return [res, resBody];
   }
 
+  /**
+   * W3C /timeouts can take as many as 3 timeout types at once, MJSONWP /timeouts only takes one
+   * at a time. So if we're using W3C and proxying to MJSONWP and there's more than one timeout type
+   * provided in the request, we need to do 3 proxies and combine the result
+   *
+   * @param {Array} timeoutObjects List of MJSONWP + W3C compatible timeouts request objects
+   */
+  getTimeoutRequestObjects (body) {
+    const {protocol} = body;
+    const timeoutObjects = [];
+    const baseObj = {
+      script: body.script,
+      pageLoad: body.pageLoad,
+      implicit: body.implicit,
+    };
+    if (protocol === W3C) {
+      for (let [type, ms] of _.toPairs(baseObj)) {
+        if (_.isNumber(ms)) {
+          timeoutObjects.push({
+            ...baseObj,
+            type,
+            ms,
+          });
+        }
+      }
+    } else {
+      timeoutObjects.push({
+        ...baseObj,
+        [body.type]: body.ms,
+        type: body.type,
+        ms: body.ms,
+      });
+    }
+    return timeoutObjects;
+  }
+
+  /**
+   * Proxy an array of timeout objects and merge the result
+   * @param {Array} timeoutObjects List of W3C+MJSONWP compatible /timeouts requests
+   */
+  async proxySetTimeouts (url, method, body) {
+    let response, resBody;
+    for (let timeoutObj of this.getTimeoutRequestObjects(body)) {
+      [response, resBody] = await this.proxy(url, method, timeoutObj);
+
+      let protocol = this.getProtocolFromResBody(resBody);
+
+      // If we got a non-MJSONWP response, return the result, nothing left to do
+      if (protocol !== MJSONWP) {
```

Comment:
I think this is redundant, since there will be anyway only one item in the array in case of mjsonwp

Reply:
If there's an array of timeout objects like this:

```javascript
[
  {pageLoad: 1, script: 2, implicit: 3, type: 'pageLoad', ms: 1},
  {pageLoad: 1, script: 2, implicit: 3, type: 'script', ms: 2},
  {pageLoad: 1, script: 2, implicit: 3, type: 'implicit', ms: 3},
]
```

...then we need to stop after the first call if it's W3C because it's done.